activerecord/lib/active_record/reflection.rb

```diff
@@ -759,6 +759,16 @@ def initialize(delegate_reflection)
         @delegate_reflection = delegate_reflection
         @klass = delegate_reflection.options[:anonymous_class]
         @source_reflection_name = delegate_reflection.options[:source]
+        if class_option = delegate_reflection.options[:class_name]
+          custom_message = class_option == source_reflection.class_name ?
+            "Remove :class_name option because it matches the source association class name" :
```

Comment:
Wording suggestion:
```suggestion
            "Remove the :class_name option because it matches the source association class name and is not needed" :
```